### PR TITLE
Add following system data layer

### LINKED ISSUE #434: Following system - Data layer
Parent: #430

## Scope
Data layer for the following system (PR 1a of 2).

### Changes
- Add `followingIds: List<String>` to UserProfile
- Add `followerIds: List<String>` to UserProfile
- Add repository methods:
  - `followUser(currentUserId, targetUserId)` - atomic Firestore transaction
  - `unfollowUser(currentUserId, targetUserId)` - atomic Firestore transaction
  - `isFollowing(currentUserId, targetUserId)` - check follow status

### Tests
- Cannot follow/unfollow yourself
- isFollowing returns correct results

## Acceptance Criteria
- [x] UserProfile has followingIds and followerIds fields
- [x] Repository methods work with Firestore transactions
- [x] Unit tests pass

### PR DESCRIPTION
**Pull request by ziyad-elarroussi**
## Description
Add data layer for the following system, enabling users to follow/unfollow other users.

**Related Issue:** Closes #434, Part of #430

---

## Changes

**Implementation:**
- Add `followingIds: List<String>` and `followerIds: List<String>` to `UserProfile`
- Add `followUser()` method with atomic Firestore transaction
- Add `unfollowUser()` method with atomic Firestore transaction
- Add `isFollowing()` method to check follow status

**Tests:**
- 7 unit tests for following logic
- Self-follow/unfollow prevention
- isFollowing with various states (empty, contains, not contains, exception)

---

## Testing
- [x] Unit tests pass
- [ ] Instrumentation tests pass
- [x] Manual testing completed
- [x] Coverage maintained (≥80%)

**Test summary:** Added 7 unit tests covering isFollowing edge cases and self-follow prevention. Transaction methods (followUser/unfollowUser) require integration tests due to Firestore transaction complexity.

---

## Checklist
- [x] Sufficient documentation and minimal inline comments
- [x] All tests passing
- [x] No new warnings
- [x] If related issue exists: issue has all labels, fields, and milestone filled

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/src/main/java/com/swent/mapin/model/UserProfileRepository.kt [hunk 1]
```diff
@@ -76,4 +76,77 @@ class UserProfileRepository(
...
+   * target user's followerIds.
+   */
+  suspend fun followUser(currentUserId: String, targetUserId: String): Boolean {
+    return updateFollowStatus(currentUserId, targetUserId, shouldFollow = true)
+  }
+
+  /**
+   * Unfollow a user. Removes targetUserId from current user's followingIds, and removes
+   * currentUserId from target user's followerIds.
+   */
```
Comment (e-bernier) [inline app/src/main/java/com/swent/mapin/model/UserProfileRepository.kt]: Same here for the Kdoc

**Review by e-bernier — CHANGES_REQUESTED** [review]
Really good job. Please try to add Kdoc as much as possible for your methods (I've pinpointed incomplete Kdoc but you could add some more to other methods). Especially since we are soon hitting the end of the project and will have to make sure it's present everywhere. Thanks !

**Review by e-bernier — APPROVED** [review]
Perfect, thank you !
